packages/core/dialect/saphana/index.ts

```diff
@@ -0,0 +1,197 @@
+  prepare(query: string, handler: (err: any, statement: Statement) => void);
+}
+
+export default class SAPHana extends GenericDialect<HanaConnection> implements ConnectionDialect {
+
+  public static deps: typeof GenericDialect['deps'] = [{
+    type: 'package',
+    name: '@sap/hana-client'
+  }];
+
+  private get lib() {
+    return __non_webpack_require__('@sap/hana-client') as HanaClientModule;
+  }
+
+  queries = queries;
+  private schema: String;
+
+  public open(encrypt?: boolean): Promise<HanaConnection> {
+    if (this.connection) {
+      return this.connection;
+    }
+
+    const connOptions = {
+      host: this.credentials.server,
+      port: this.credentials.port,
+      user: this.credentials.username,
+      password: this.credentials.password
+    };
+    try {
+      let conn = this.lib.createClient(connOptions);
+
+      this.connection = new Promise<HanaConnection>((resolve, reject) => conn.connect(err => {
+        if (err) {
+          console.error("Connection to HANA failed", err.toString());
+          reject(err);
+        }
+        this.schema = this.credentials.database;
+        conn.exec("SET SCHEMA " + this.schema, err => {
+            if (err) {
+              reject(err);
+            }
+            console.log("Connection to SAP Hana succedded!");
+            resolve(conn);
+          });
+      }));
+      return this.connection;
+    } catch (e) {
+      console.error("Connection to HANA failed" + e.toString());
+      Promise.reject(e);
+    }
+  }
+
+  public async close() {
+    if (!this.connection) return Promise.resolve();
+
+    await this.connection.then(conn => conn.disconnect());
+    this.connection = null;
+  }
+
+  public async testConnection?() {
+    return this.open().then(conn => conn.exec('select 1 from dummy;', (err, rows) => rows));
+  }
+
+  public query(query: string, args?): Promise<DatabaseInterface.QueryResults[]> {
+    return this.open().then(conn => {
+      return new Promise<DatabaseInterface.QueryResults[]>((resolve, reject) => {
+        if (args) {
+          conn.prepare(query, (err, statement) => {
+            if (err) {
+              return this.resolveErr(resolve, err, query);
+            }
+            statement.exec(args, (err, rows) => {
+              if (err) {
+                return this.resolveErr(resolve, err, query);
+              }
+              return this.resolveQueryResults(resolve, rows, query);
+            });
+          });
+        } else {
+          conn.exec(query, (err, rows) => {
+            if (err) {
+              return this.resolveErr(resolve, err, query);
+            }
+            return this.resolveQueryResults(resolve, rows, query);
+          });
+        }
+      });
+    });
+  }
+
+  private resolveQueryResults(resolve, rows, query) {
+    let cols: string[] = [];
+    if (rows && rows.length > 0) {
+      for (let colName in rows[0]) {
+        cols.push(colName);
+      }
+    }
+
+    let res = {
+      connId: "1",
```

Comment:
```suggestion
      connId: this.getId(),
```

This is the conn id generated by extension. It's used for some commands like exporting results